janus.c

```diff
@@ -191,16 +191,33 @@ static gchar *local_ip = NULL;
 gchar *janus_get_local_ip(void) {
 	return local_ip;
 }
-static gchar *public_ip = NULL;
-gchar *janus_get_public_ip(void) {
-	/* Fallback to the local IP, if we have no public one */
-	return public_ip ? public_ip : local_ip;
+static GHashTable *public_ips_table = NULL;
+static GList *public_ips = NULL;
+guint janus_get_public_ip_count(void) {
+	return public_ips_table ? g_hash_table_size(public_ips_table) : 0;
 }
-void janus_set_public_ip(const char *ip) {
-	/* once set do not override */
-	if(ip == NULL || public_ip != NULL)
+gchar *janus_get_public_ip(guint index) {
+	if (!janus_get_public_ip_count()) {
+		/* Fallback to the local IP, if we have no public one */
+		return local_ip;
+	}
+	if (index >= g_hash_table_size(public_ips_table)) {
+		return NULL;
```

Comment:
I don't think we should ever return NULL. Many places in the code assume it's a valid string (e.g., doing g_strdup), and would crash in case. Maybe just cap `index` to `g_hash_table_size(public_ips_table)-1` if it's too big?

Reply:
Done